### PR TITLE
Improve extendability in sale and account

### PR DESCRIPTION
**Pull request by davidtranhp**
This PR aims to separate vals preparation into new method for hooking in extensions

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_bank_statement.py [hunk 1]
```diff
@@ -670,14 +687,12 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
...
+                bank_account = self.env['res.partner.bank'].create(self._prepare_partner_bank_data())
             self.bank_account_id = bank_account
 
         counterpart_moves.assert_balanced()
         return counterpart_moves
 
     def _check_invoice_state(self, invoice):
         if invoice.state == 'in_payment' and all([payment.state == 'reconciled' for payment in invoice.mapped('payment_move_line_ids.payment_id')]):
-           invoice.write({'state': 'paid'})
+            invoice.write({'state': 'paid'})
```
Comment (gdgellatly) [inline addons/account/models/account_bank_statement.py]: Wrong

File: addons/sale/models/sale.py [hunk 2]
```diff
@@ -676,18 +676,26 @@ def action_confirm(self):
     def _get_forbidden_state_confirm(self):
         return {'done', 'cancel'}
 
+    def _prepare_analytic_account_data(self, prefix=None):
```
Comment (gdgellatly) [inline addons/sale/models/sale.py]: Need to seperate PR's I would have thought. Pretty rare for a PR doing 2 unrelated things to 2 modules to be accepted but see how you go.

File: addons/account/models/account_bank_statement.py (hunk 1 above)
Comment (gdgellatly) [reply addons/account/models/account_bank_statement.py]: actually right, oops. my eyes are wrong. Thought it had gone 5 spaces.

File: addons/sale/models/sale.py (hunk 2 above)
Comment (davidtranhp) [reply addons/sale/models/sale.py]: I was on my way to create more commits to improve extendability but just stopped to see if I would get my luck.

**Comment by mart-e** [comment]
Sorry for the lack of followup on this PR.
The code has changed a lot now, it seems there is an hook method for the payment data in master

https://github.com/odoo/odoo/blob/0fe2b4530f97de35a7aa78beb6265f4a5dcef55d/addons/account/models/account_bank_statement.py#L603

Let's close this one and if you still need different hook, please open a new PR and ping me, I promise I will review it faster !

**Comment by davidtranhp** [comment]
@mart-e Tks for your attention. This is another new PR https://github.com/odoo/odoo/pull/35784 containing a part of this for your review.

**Pull request closed without merging**